rasa/core/policies/ensemble.py

```diff
@@ -119,59 +119,82 @@ def _check_priorities(self) -> None:
-            else:
-                ml_trackers.append(tracker)
-
-        return ml_trackers, rule_trackers
-
-    def _policy_ensemble_contains_rule_policy(self) -> bool:
-        """Determine whether `RulePolicy` is part of the policy ensemble."""
-        return any(isinstance(policy, RulePolicy) for policy in self.policies)
+        return any(tracker.is_rule_tracker for tracker in training_trackers)
 
     def _emit_rule_policy_warning(
-        self, rule_trackers: List[DialogueStateTracker]
+        self, training_trackers: List[DialogueStateTracker]
     ) -> None:
-        """Emit `UserWarning`s about missing `RulePolicy` training data."""
-        rule_policy_is_active = self._policy_ensemble_contains_rule_policy()
+        """Emit `UserWarning`s about missing rule-based data."""
+        rule_policy_is_active = (
+            self._policy_ensemble_contains_policy_with_rules_support()
+        )
+        training_trackers_contain_rule_trackers = self._training_trackers_contain_rule_trackers(
+            training_trackers
+        )
 
         # TODO: add new docs links to these warnings
-        if rule_policy_is_active and not rule_trackers:
+        if rule_policy_is_active and not training_trackers_contain_rule_trackers:
             common_utils.raise_warning(
-                f"Found `{RulePolicy.__name__}` in your pipeline but "
+                f"Found a rule-based policy in your pipeline but "
                 f"no rule-based training data. Please add rule-based "
                 f"stories to your training data or "
-                f"remove the `{RulePolicy.__name__}` entry in "
+                f"remove the rule-based policy (`{RulePolicy.__name__}`) from your "
                 f"your pipeline."
             )
-        elif not rule_policy_is_active and rule_trackers:
+        elif not rule_policy_is_active and training_trackers_contain_rule_trackers:
             common_utils.raise_warning(
-                f"Found rule-based training data but `{RulePolicy.__name__}` "
-                f"is not part of the configuration. Please add "
-                f"`{RulePolicy.__name__}` to the "
-                f"`policies` section in `{DEFAULT_CONFIG_PATH}`."
+                f"Found rule-based training data but no policy supporting rule-based "
+                f"data. Please add `{RulePolicy.__name__}` or another rule-supporting "
+                f"policy to the `policies` section in `{DEFAULT_CONFIG_PATH}`."
             )
 
+    @staticmethod
+    def _get_training_trackers_for_policy(
+        policy: Policy, training_trackers: List[DialogueStateTracker]
+    ) -> List[DialogueStateTracker]:
+        """Return trackers to be trained on ML-based policies and `RulePolicy`.
+
+        Args:
+            training_trackers: Trackers to split.
+
+        Returns:
+            Trackers from ML-based training data and rule-based data.
+        """
+        supported_data = policy.supported_data()
```

Comment:
optional: I'm wondering if splitting the training data should actually be the responsibility of the `Ensemble` or rather some function / static method with `trackers` or even a method of `SupportData` 🤔

Reply:
👍  good point, moved it to `SupportedData`